tests/conftest.py

```diff
@@ -47,13 +59,257 @@ def invalid_cephfs_storageclass(request):
     storageclass = OCS(**yaml_data)
     sc_data = storageclass.create()
 
-    logger.debug('Check that storageclass has assigned creationTimestamp')
+    log.debug('Check that storageclass has assigned creationTimestamp')
     assert sc_data['metadata']['creationTimestamp']
 
     yield sc_data
 
-    logger.info(
+    log.info(
         f"TEARDOWN - removing storageclass "
         f"{request.param['storageclass_name']}"
     )
     storageclass.delete()
+
+
+def cluster_teardown():
+    log.info("Destroying the test cluster")
+    destroy_cluster(ENV_DATA['cluster_path'])
+    log.info("Destroying the test cluster complete")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def cluster(request):
+    log.info("Running OCS basic installation")
+    cluster_path = ENV_DATA['cluster_path']
+    # Add a finalizer to teardown the cluster after test execution is finished
+    if config.teardown:
+        request.addfinalizer(cluster_teardown)
+        log.info("Will teardown cluster because --teardown was provided")
+    # Test cluster access and if exist just skip the deployment.
+    if is_cluster_running(cluster_path):
+        log.info("The installation is skipped because the cluster is running")
+        return
+    elif config.teardown and not config.deploy:
+        log.info("Attempting teardown of non-accessible cluster: %s", cluster_path)
+        return
+    elif not config.deploy and not config.teardown:
+        msg = "The given cluster can not be connected to: {}. ".format(cluster_path)
+        msg += "Provide a valid --cluster-path or use --deploy to deploy a new cluster"
+        pytest.fail(msg)
```

Comment:
It may be nice to fail when both teardown and deploy are specified. It looks like this will just run teardown if cluster is running, and deploy and teardown if cluster exists, but is not accessible.

Reply:
I was able to run teardown on a cluster that did not pass the accessibility check. You'll need to be able to specify ``--teardown`` and ``--deploy`` if you want to spin up a testing cluster and immediately destroy it after testing. I think this is something we should allow.